### ISSUE TITLE
Create trusted image CRD

### ISSUE THREAD

**Issue by JorritSalverda** [estafette/estafette-ci-operator#2]
Support a trustedimage CRD with the following api version and kind

```yaml
apiVersion: ci.estafette.io/v1
kind: trustedimage
```

And the following fields in the spec:

```yaml
path: <string, required and needs to be unique>
runPrivileged: <bool, optional>
runDocker: <bool, optional>
allowCommands: <bool, optional>
injectedCredentialTypes: <[]string, optional>
whitelistedPipelines: <string, optional>
```

The path is a path to a container image including repository, but excluding the tag.

See https://github.com/estafette/estafette-ci-contracts/blob/master/config.go#L81-L89

**Comment by vitorarins** [comment]
Do we need `type` and `whitelistedTrustedImages`?

**Comment by JorritSalverda** [comment]
Yes `type` is used when injecting them in trusted images, for examples `kubernetes-engine` or `container-registry`. The `whitelistedTrustedImages` is to prevent some credentials getting used for any trusted image that wants to make use of it. For example you wouldn't want the port-forward or gke-remote-stage extension to get access to production cluster credentials.

**Comment by vitorarins** [comment]
I'm confused here, seems like you are talking about the credential CRD but this is the trusted image CRD.

**Comment by JorritSalverda** [comment]
Sorry I didn't read properly the last time, since this is the issue for trusted image CRD. I'll update the issue. Copy/paste error.

**Comment by vitorarins** [comment]
Fixed by #5
